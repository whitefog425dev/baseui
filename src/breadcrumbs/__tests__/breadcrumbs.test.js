/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledLink} from '../../link/index.js';
import {Breadcrumbs} from '../index.js';

describe('Breadcrumbs', () => {
  it('applies correct accessibility attributes to root element', () => {
    const example = shallow(
      <Breadcrumbs>
        <StyledLink href="#">Parent Page</StyledLink>
        <StyledLink href="#">Sub-Parent Page</StyledLink>
        <span>Current Page</span>
      </Breadcrumbs>,
    );
    expect(example).toHaveProp('aria-label', 'Breadcrumbs navigation');
  });

  it('displays separators in correct positions', () => {
    expect(
      shallow(
        <Breadcrumbs>
          <StyledLink href="#">Parent Page</StyledLink>
          <StyledLink href="#">Sub-Parent Page</StyledLink>
          <span>Current Page</span>
        </Breadcrumbs>,
      ),
    ).toMatchSnapshot();
  });
});
