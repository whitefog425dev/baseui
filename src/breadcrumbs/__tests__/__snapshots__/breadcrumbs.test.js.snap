// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs displays separators in correct positions 1`] = `
<MockStyledComponent
  aria-label="Breadcrumbs navigation"
>
  <MockStyledComponent
    href="#"
  >
    Parent Page
  </MockStyledComponent>
  <MockStyledComponent
    key="0"
  >
    <MockStyledComponent />
  </MockStyledComponent>
  <MockStyledComponent
    href="#"
  >
    Sub-Parent Page
  </MockStyledComponent>
  <MockStyledComponent
    key="1"
  >
    <MockStyledComponent />
  </MockStyledComponent>
  <span>
    Current Page
  </span>
</MockStyledComponent>
`;
